src/plugins/plugin-express.js

```diff
@@ -0,0 +1,93 @@
+/**
+ * Copyright 2017 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict';
+var shimmer = require('shimmer');
+var methods = require('methods').concat('use', 'route', 'param', 'all');
+
+var SUPPORTED_VERSIONS = '4.x';
+
+function patchModuleRoot(express, api) {
+  var labels = api.labels;
+  function applicationActionWrap(method) {
+    return function expressActionTrace() {
+      if (!this._google_trace_patched && !this._router) {
+        this._google_trace_patched = true;
+        this.use(middleware);
+      }
+      return method.apply(this, arguments);
+    };
+  }
+
+  function middleware(req, res, next) {
+    var options = {
+      name: req.path,
+      traceContext: req.get(api.constants.TRACE_CONTEXT_HEADER_NAME),
```

Comment:
`traceContext` now accepts a string, which it parses. It doesn't parse it if it's not a string, so it's OK for `req.get` to return null.